library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -133,8 +134,10 @@ protected void transferWindowFiles()
           Map<Long, Map<Slice, Bucket.BucketedValue>> buckets = savedWindows.remove(windowId);
 
           for (Map.Entry<Long, Map<Slice, Bucket.BucketedValue>> singleBucket : buckets.entrySet()) {
-            managedStateContext.getBucketsFileSystem().writeBucketData(windowId, singleBucket.getKey(),
-                singleBucket.getValue());
+            long bucketId = singleBucket.getKey();
+            if (bucketId > latestPurgedTimeBucket) {
```

Comment:
The intention here is to avoid transferring buckets that are **already** purged. Without this condition, my test run throws an exception complaining that the bucket it tries to transfer does not exist. I will add a unit test for this, but this looks like it's causing another unit test failure in ManageStateImplTest.testFreeWindowTransferRaceCondition so I wanted to make sure what I'm doing is correct first.

Did you mean `latestPurgedBucketId`? If this variable is not transient, the value will not be valid after recovery, and it would be possible that it would still try to transfer purged buckets after recovery. I'm not exactly familiar with the Managed State code so correct me if I'm wrong.